repository_test.go

```diff
@@ -1307,6 +1307,39 @@ func TestRepository_Update(t *testing.T) {
 	adapter.AssertExpectations(t)
 }
 
+func TestRepository_Update_ptrElem(t *testing.T) {
+	var (
+		adapter = &testAdapter{}
+		repo    = New(adapter)
+		user    = &User{
+			ID:        1,
+			Name:      "name",
+			CreatedAt: Now(),
+			UpdatedAt: Now(),
+		}
+		mutates = map[string]Mutate{
+			"id":         Set("id", 1),
+			"name":       Set("name", "name"),
+			"age":        Set("age", 0),
+			"created_at": Set("created_at", Now()),
+			"updated_at": Set("updated_at", Now()),
+		}
+		queries = From("users").Where(Eq("id", user.ID))
+	)
+
+	adapter.On("Update", queries, "id", mutates).Return(1, nil).Once()
+
+	assert.Nil(t, repo.Update(context.TODO(), &user))
```

Comment:
this case is double pointer? 🤔

Reply:
I'm not really sure if we should handle this case, since this case can be further extended to multiple level of pointers 🤔

Reply:
yes, sorry, completely wrong test case 😅